### PR TITLE
gatekeep mdns ipv6 behind feature flag

### PR DESCRIPTION
**Pull request by indexds**
Hello,

This is a very ugly hack so you most likely do not want to merge this as is, i'm open to working on any modifications.
The point of the commit is that currently, if ipv6 is disabled in sdkconfig and mdns is imported as an external component in Cargo.toml, the project does not compile because the C functions called by mdns.rs aren't exposed anymore.

Is there a better way to do this? In the meantime, this worked for me.

### CODE REVIEW

**Review by ivmarkov — CHANGES_REQUESTED** [review]

File: Cargo.toml [hunk 1]
```diff
@@ -23,7 +23,9 @@ esp-idf-hal = { git = "https://github.com/esp-rs/esp-idf-hal" }
 esp-idf-sys = { git = "https://github.com/esp-rs/esp-idf-sys" }
 
 [features]
-default = ["std", "binstart"]
+default = ["std", "binstart", "mdns_ipv6"]
+
+mdns_ipv6 = []
```
[the remark this answers is not in the document]
Comment (indexds) [reply Cargo.toml:28]: > Not necessary, remove.
> 
> Assuming you are using [CONFIG_LWIP_IPV6](https://docs.espressif.com/projects/esp-idf/en/stable/esp32/api-reference/kconfig.html#config-lwip-ipv6) in your `sdkconfig.defaults`, you can just do `#[cfg(esp_idf_lwip_ipv6)]` in the code to conditionally compile-out IPv6 support.

That makes sense, should I add that cfg for ipv4 as well for the sake of symmetry? Some people only do ipv6 I believe.

[the remark this answers is not in the document]
Comment (ivmarkov) [reply Cargo.toml]: Sure and why not? Just don't ne surprised if `esp-idf-svc` does not compile after you put the no-ipv4 conf, as I'm sure there are (a lot of) other places which we might need to protect if ipv4 is disabled.

**Comment by indexds** [comment]
This should be fine, lmk

**Comment by ivmarkov** [comment]
Thanks!

**Pull request merged**